napari/plugins/hook_specifications.py

```diff
@@ -317,3 +328,53 @@ def napari_write_vectors(path: str, data: Any, meta: dict) -> Optional[str]:
         If data is successfully written, return the ``path`` that was written.
         Otherwise, if nothing was done, return ``None``.
     """
+
+
+# -------------------------------------------------------------------------- #
+#                                 GUI Hooks                                  #
+# -------------------------------------------------------------------------- #
+
+
+@napari_hook_specification(historic=True)
```

Comment:
What does historic mean? I found [this explanation](https://github.com/napari/napari/issues/1272#issuecomment-633064212) by @tlambert03, together with a link to the [pluggy docs on it](https://pluggy.readthedocs.io/en/latest/index.html#historic-hooks), but it is still a mysterious term to me:

> (call historic means that the implementation will be called once immediately on registration, regardless of whether it is registered before or after the hook specification)

I am wondering whether we should rename it to e.g. `call_immediately=`, because `historic=` is clear as molasses. My brain goes to either deprecated hookspecs or hookspecs that provide undo behaviour.

Reply:
I think "historic" happens once and doesn't get changed, so I'm ok with the name. I don't have strong feelings, but maybe a mild preference to not deviating from pluggy names unless it is a last resort. I'll defer to @tlambert03. That would need to be a new breaking change for the napari plugin engine too, and not in this PR

Reply:
i'm also ok with the name.  maybe wouldn't have picked it personally (and also wouldn't oppose a PR if someone really wanted to change it...), but it doesn't feel urgent to me.  The problem with `call_immediately` is that it's not really called immediately.  the `plugin_manager` gets to "call" the hookspec, and even if all the plugins haven't yet registered their implementations, when they _later_ get registered, they will then be able to provide results (like a list of widgets) for the hookspec that was called by the plugin manager in the past ("historically").

I feel like this explanation from the pluggy docs mostly covers it?

> The implication is that late registered hookimpls will be called back immediately at register time and can not return a result to the caller. This turns out to be particularly useful when dealing with lazy or dynamically loaded plugins.

Reply:
> it's not really called immediately

oh ... I guess the plugin is called immediately :) not the hook call.  ah relativism

Reply:
> I feel like this explanation from the pluggy docs mostly explains it?

I still don't grok the link to the word "historic", but maybe I just need to brush up on my grammar? OED definition:

> (grammar) (of a tense) used in the narration of past events, especially Latin and Greek imperfect and pluperfect.

Agreed that this isn't urgent — this terminology doesn't leak into user/plugin-writer space, does it? In which case I'm happy to punt.

Reply:
> this terminology doesn't leak into user/plugin-writer space, does it?

I don't think it does. It's all internal to napari